discord_bot/discord_bot.py

```diff
@@ -34,53 +36,94 @@ def load_config():
         return r.json()['url']
 
     async def send_json(self, payload):
         asyncio.ensure_future(self.websocket.send(json.dumps(payload)))
 
     async def handshake(self, message):
         """
-        When a websocket connection is opened, Hello payload is received. Then, an Identify payload is sent as
+        When a websocket connection is opened, Hello payload is received. Then, an Identify/Resume payload is sent as
         part of the handshake to authorize this client.
 
         :param message: dict - hello payload
         """
-        self.heartbeat_interval_ms = message["d"]["heartbeat_interval"]
+        self.heartbeat_interval_ms = message["d"]["heartbeat_interval"]  # seconds
         asyncio.ensure_future(self.heartbeat())
-
         handshake_identity = self.config["handshake_identity"]
-        asyncio.ensure_future(self.send_json({"op": Opcodes.IDENTIFY, "d": handshake_identity}))
+        if not self.session_id:
+            asyncio.ensure_future(self.send_json({"op": Opcodes.IDENTIFY, "d": handshake_identity}))
+        else:
+            asyncio.ensure_future(self.send_json({"op": Opcodes.RESUME,
+                                                  "d": {"token": self.config["handshake_identity"]["token"],
+                                                        "session_id": self.session_id,
+                                                        "seq": self.last_seq}}))
 
     async def heartbeat(self):
         """
-        Sends a heartbeat payload every heartbeat interval.
+        Sends a heartbeat payload every heartbeat interval and allows messages to be sent.
         """
         await asyncio.sleep(self.heartbeat_interval_ms / 1000.0)
+        print("Last Sequence: {}".format(self.last_seq))
         asyncio.ensure_future(self.send_json({"op": Opcodes.HEARTBEAT, "d": self.last_seq}))
         asyncio.ensure_future(self.heartbeat())
+        # asyncio.ensure_future(self.send_message("embed": {
+        #                                             "title": ("{data[repository][owner_name]}/"
+        #                                                       "{data[repository][name]} "
+        #                                                       "{data[status_message]}"
+        #                                                       ).format(data=data),
+        #                                             "type": "rich",
+        #                                             "description": ("{data[author_name]} {data[type]} "
+        #                                                             "<{data[compare_url]}>"
+        #                                                             ).format(data=data),
+        #                                             "url": data['build_url']}))
 
     async def gateway_handler(self):
         """
-        Creates the websocket, receives messages and acts on them.
+        Creates the websocket, receives responses and acts on them.
```

Comment:
These aren't really responses, they are payloads sent by the server. A response is specifically something sent in response to a request, usually referring to HTTP requests/responses. "Message" is a more proper term.